storage_controller/src/service.rs

```diff
@@ -3609,6 +3609,13 @@ impl Service {
                     }
                 }
 
+                // Ensure shard still complies with its policy (e.g. number of secondaries)
+                if let Err(e) = shard.schedule(scheduler, &mut ScheduleContext::default()) {
+                    // This might happen if we are migrating and there are not enough nodes for
+                    // secondaries, for example
+                    tracing::warn!("Scheduling error during migration: {e}");
+                }
```

Comment:
It's a bit odd that we do manual scheduling above and then call into the scheduler just to validate/fixup. The code above should maintain the count of secondaries, so not sure I get the `e.g.`. What was the actual issue?

Reply:
It is indeed... I backtracked a bit and realized that this was working around a bug that didn't exist on this branch (on the branch where I had been working on this, I had made a mistake in a different commit that resulted in an extra secondary sometimes.)

So this was harmless, but redundant.  Reverted this change.